### PR TITLE
OT275 27: Validación de acceso en base a roles asociados a los endpoints

### PR DESCRIPTION
**Pull request by Sieglind**
# [OT275-27](https://alkemy-labs.atlassian.net/browse/OT275-27)

Created AuthenticationEntryPoint and added it to WebSecurityConfigurerAdapter#configure
Created response for access attempts to resources without proper credentials

### HOW HAS THIS BEEN TESTED?

Please describe the tests that you ran to verify your changes. Provide instructions so we can
reproduce. Please also list any relevant details for your test configuration.

- [x] Postman.
- [ ] Unit test.
- [ ] Integration test.
- [ ] No testing required (only applies for tech task).

**Test Configuration**:

* ...
* ...

**Screenshots**:
When reaching and endpoint without proper credentials or being authenticated, the response we get is:
image

* ....

### CHECKLIST

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have added the new resource in the Postman Collection file.

### CODE REVIEW

File: src/main/java/com/alkemy/ong/infrastructure/config/spring/security/filter/AuthEntryPoint.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+    OutputStream responseStream = response.getOutputStream();
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.writeValue(responseStream,
+        new AuthenticationError(new Timestamp(System.currentTimeMillis()),
+            request.getServletPath()));
+    responseStream.flush();
+  }
+
+
+  private static class AuthenticationError {
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/config/spring/security/filter/AuthEntryPoint.java:34]: REUTILIZAR la class ErrorResponse.

File: src/main/java/com/alkemy/ong/infrastructure/config/spring/security/filter/AuthEntryPoint.java [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+    public Timestamp timestamp;
+    public final int statusCode = 403;
+    public final String error = "Forbidden";
+    public final String message = "Invalid Credentials";
```
Comment (mkain0) [inline src/main/java/com/alkemy/ong/infrastructure/config/spring/security/filter/AuthEntryPoint.java:39]: TRIVIAL: pondría algo como "Access denied. Contact your administrator."

**Review by mkain0 — CHANGES_REQUESTED** [review]

**Comment by Sieglind** [comment]
Hola Magali, las peticiones de cambio que se denotan como "TRIVIAL", queda a nuestra elección implementarlas?

**Review by mkain0 — APPROVED** [review]

**Comment by mkain0** [comment]
> Hola Magali, las peticiones de cambio que se denotan como "TRIVIAL", queda a nuestra elección implementarlas?

Exacto.

**Pull request merged**